### ISSUE TITLE
During CPU training, validation accuracy is always much smaller than training accuracy

### ISSUE THREAD

**Issue by lidanqing-intel** [PaddlePaddle/PaddleClas#486]
Hi, @luotao1 

We enabled resnet50 and se-resnext oneDNN dygraph training in models repo https://github.com/PaddlePaddle/models, both achieved good accuracy.
Since `models` repo will be deprecated soon, we move to `PaddleClas` repo recently, but we found that using the yaml and scripts in PaddleClas, the training accuracy and validation accuracy has big gap. In detail, after some epoches of training, resnet50 and se-resnext dygraph training accuracy improve as expected, but validation accuracy is much lower than training accuracy.

We do not understand the cause of the accuracy gap between training accuracy and validation. In theory, models repo and PaddleClas repo only provide training script, operators are all from Paddle repo. But it turn out models repo run well but PaddleClas repo have big training and validation accuracy gap.

Could you please give some suggestions? Thank you very much !

@arlesniak could provide the log of training.

**Comment by littletomatodonkey** [comment]
1. What is the dataset you used for training, if it is the flowers102 dataset, you may load the pretrained model and finetune. The distribution of the training and validation dataset should also be similar.
2. If the dataset is too small, then overfitting may be a problem.
3. By the way,  could you provide the config file in your training process?

**Comment by arlesniak** [comment]
Regarding se_resnext
1. Dataset we use is flowers102 , we train from the scratch to compare the results with models repo.
2. You're right, we expect the possibility of overfitting however in the log the validation acc1 looks like it stucked around 0.36 (since epoch 39) and training is increasing acc1 0.79 (now epoch 54).
3. cmd line is as follows:
/usr/bin/python2.7 tools/train.py -c configs/ResNeXt/ResNeXt50_32x4d.yaml -o ARCHITECTURE.name=SE_ResNeXt50_32x4d -o pretrained_model= -o TRAIN.file_list=dataset/flowers102/train_list.txt -o TRAIN.data_dir=dataset/flowers102 -o VALID.file_list=dataset/flowers102/val_list.txt -o VALID.data_dir=dataset/flowers102 -o use_gpu=False -o validate=True -o classes_num=102 -o total_images=1020

Paddle branch develop commit 81acc3278c36f0fdfda48f13947f3236b3f616a4 (14 Dec)
PaddleClas branch dygraph

We expect acc1 to be higher, could you give some suggestions? Thank you very much !

**Comment by arlesniak** [comment]
se_resnext current log with verbose config (trainig is still running) :
[train_se_resnext_cpu_flowers_with_val.log](https://github.com/PaddlePaddle/PaddleClas/files/5704038/train_se_resnext_cpu_flowers_with_val.log)

**Comment by littletomatodonkey** [comment]
1. L2 decay should be larger for the small dataset(you can try 0.001 or 0.0005 )

https://github.com/PaddlePaddle/PaddleClas/blob/fdfce2d6d5439639b1111db101deea4a5730408c/configs/SENet/SE_ResNeXt50_vd_32x4d.yaml#L30

2. More augmentation methods are worth trying to avoid over-fitting, such as auto-augmentation, random augmentation and random erasing, 

https://github.com/PaddlePaddle/PaddleClas/blob/fdfce2d6d5439639b1111db101deea4a5730408c/configs/DataAugment/ResNet50_AutoAugment.yaml#L47

https://github.com/PaddlePaddle/PaddleClas/blob/fdfce2d6d5439639b1111db101deea4a5730408c/configs/DataAugment/ResNet50_RandAugment.yaml#L47

https://github.com/PaddlePaddle/PaddleClas/blob/fdfce2d6d5439639b1111db101deea4a5730408c/configs/DataAugment/ResNet50_RandomErasing.yaml#L53

3. batch size and learning rate may be decreased proportionally, such as batch size for 64 and learning rate for 0.025 for more mini-batches in one epoch (Smaller batch size also takes more variance for batch norm statistics, which may alleviate  the effect of over-fitting).

4. Since the dataset is so small, maybe longer training epochs is a better choice, you can try [50,100,150] for piecewise decay epoch and check whether it is useful for the val metric.

**Comment by arlesniak** [comment]
Thank you for the info, we'll try the suggestions for **se_resnext**.

We have questions also about **Resnet50** PaddleClas training. 
We used values for hyperparams as in models repo, regarding batch=32 etc :
https://github.com/PaddlePaddle/models/blob/develop/dygraph/resnet/train.py#L34
I trains succesfully after 120 epochs to:
final eval acc1 0.698 acc5 0.902

However in PaddleClas  after 120 epochs we got:
END epoch:119 valid loss: 3.38436, top1: **0.19706**, top5: **0.48725**
INFO: The best top1 **acc 0.20098**, in epoch: 85

We used configs/ResNet/ResNet50.yaml and ARCHITECTURE.name=ResNet50
current log with verbose config : [train_resnet50_cpu_b32_flowers_with_val.log](https://github.com/PaddlePaddle/PaddleClas/files/5708141/train_resnet50_cpu_b32_flowers_with_val.log)

Could you give some suggestions? Thank you very much !

**Comment by CZH-TOP** [comment]
Thanks for your questions! 
We found that you did not use train_extra_list.txt in training stage on PaddleClas!
`-o TRAIN.file_list=dataset/flowers102/train_list.txt`-->`-o TRAIN.file_list=dataset/flowers102/train_all_list.txt`
train_all_list consists of train_list and train_extra_list.
configs at:[train_resnet50_gpu_b32_flowers_with_val.log](https://github.com/PaddlePaddle/PaddleClas/files/5741582/train_resnet50_gpu_b32_flowers_with_val.log)
Best wishes!

**Comment by arlesniak** [comment]
@CZH-TOP Thank you for the suggestion. We'll try that.

**Comment by littletomatodonkey** [comment]
OK, feel free to reopen the issue or open a new issue if there are other problems~
